src/mlpack/bindings/cli/add_to_po.hpp

```diff
@@ -115,10 +78,36 @@ void AddToPO(util::ParamData& param,
   std::string cliName = (param.alias != '\0') ?
       "-" + std::string(1, param.alias) + ",--" + mappedName : "--" + mappedName;
 
+  if constexpr (std::is_same<typename ParameterType<
+                             typename std::remove_pointer<T>::type>::type, 
+                bool>())
+  {
+    app->add_flag_function(cliName.c_str(),
+        [&param](const typename ParameterType<
+                       typename std::remove_pointer<T>::type>::type& value)
+        {
+          param.value = value;
+          param.wasPassed = true;
+        },
+        param.desc.c_str());
+  }
+  else
+  {
+    app->add_option_function<typename ParameterType<
+                             typename std::remove_pointer<T>::type>::type>(cliName.c_str(),
+          [&param](const typename ParameterType<
+                         typename std::remove_pointer<T>::type>::type& value)
+          {
+           param.value = value;
+           param.wasPassed = true;
+          },
+          param.desc.c_str());
+  }
+
```

Comment:
@rcurtin I have refactored these functions as follows. I had to use `if constexpr` in order to make it compile (which is only available in C++17), otherwise we will have to keep the two functions as before.
I had referenced to this solution [here](https://stackoverflow.com/questions/48649953/stdis-same-not-working-with-bool) since I got similar errors.